src/scheduler.rs

```diff
@@ -152,10 +155,12 @@ async fn fetch_idle(ctx: &Context, connection: &mut Imap, folder: Config) -> Int
                 warn!(ctx, "{:#}", err);
             }
 
-            if let Err(err) = connection.scan_folders(&ctx).await {
-                // Don't reconnect, if there is a problem with the connection we will realize this when IDLEing
-                // but maybe just one folder can't be selected or something
-                warn!(ctx, "{}", err);
+            if folder == Config::ConfiguredInboxFolder {
```

Comment:
A comment that it is to prevent parallel scans and duplicate messages would be nice here.